controllers/llamastackdistribution_controller.go

```diff
@@ -191,9 +209,50 @@ func (r *LlamaStackDistributionReconciler) SetupWithManager(mgr ctrl.Manager) er
 		Owns(&corev1.Service{}).
 		Owns(&networkingv1.NetworkPolicy{}).
 		Owns(&corev1.PersistentVolumeClaim{}).
+		Watches(
+			&corev1.ConfigMap{},
+			handler.EnqueueRequestsFromMapFunc(r.findLlamaStackDistributionsForConfigMap),
+			builder.WithPredicates(predicate.ResourceVersionChangedPredicate{}),
+		).
 		Complete(r)
 }
 
+// findLlamaStackDistributionsForConfigMap maps ConfigMap changes to LlamaStackDistribution reconcile requests.
+func (r *LlamaStackDistributionReconciler) findLlamaStackDistributionsForConfigMap(ctx context.Context, configMap client.Object) []reconcile.Request {
+	attachedLlamaStacks := &llamav1alpha1.LlamaStackDistributionList{}
```

Comment:
No immediate action, can be done in a followup, up to you. Can you use a Field Indexer with a FieldSelector in the List calls? This will drastically improve performance on a large number of CM which is not uncommon on a cluster.

Something like:

```go
r.List(ctx, &matchedDistributions, &client.ListOptions{
		FieldSelector: client.MatchingFields{
			".spec.server.userConfig.configMapKey": key,
		},
	})
```

but you need to setup an indexer first in the manager, see: https://book.kubebuilder.io/cronjob-tutorial/controller-implementation#setup

Reply:
I started implementing this approach, but hit some errors around failing to list when using the field selector. I'll look into it more.

Reply:
Found the cause of the errors that I saw, it was due to missing selectablefield kubebuilder markers. I've made the necessary updates and will work on getting those changes included in this PR.